incubator/validator-core/src/main/java/io/aklivity/zilla/runtime/validator/core/IntegerValidator.java

```diff
@@ -26,21 +27,23 @@ public IntegerValidator(IntegerValidatorConfig config)
     }
 
     @Override
-    public DirectBuffer read(
+    public int read(
         DirectBuffer data,
         int index,
-        int length)
+        int length,
+        ToIntValueFunction next)
     {
-        return validate(data, index, length) ? data : null;
+        return !validate(data, index, length) ? -1 : next.applyAsInt(data, index, length);
```

Comment:
This one is all good to return the progress from `next` because there is no transformation, so the progress in `next` is equivalent to the progress in this validator.